QQE-379.md

```diff
@@ -0,0 +1,44 @@
+# QQE-379 - HTTP module / REST use-cases - Development topics - JSON payload
+## Links
+
+Jira : [QQE-379](https://issues.redhat.com/browse/QQE-379)
+TD:
+
+This test plan outlines the testing for different JSON payloads in Quarkus REST endpoints.
+It aims to ensure the correct handling of JSON data in various scenarios, including successful processing, error handling, and complex data structures.
+
+## Scope of the testing
+The test coverage will be verified for JVM, native mode and Openshift.
+
+### What to test
+
+ - Successful JSON payload processing: Verify that valid JSON payloads requests to REST endpoints and assert that the responses contain the correct JSON data and status codes.
+ - Check Big JSON payload processing: Verify that the application can efficiently process large JSON payloads without issues.
+ - Handling invalid JSON payloads: Test how the application responds to malformed or invalid JSON requests,
+ensuring appropriate error responses are returned.
+ - Handling missing or incorrect data types: Validate the behavior when required JSON fields are missing or have incorrect data types,
+ensuring proper error handling.
+ - Testing complex JSON structures: Evaluate the handling of JSON payloads containing nested objects, arrays, and various data types,
+ensuring serialization and deserialization.
```

Comment:
I think all those are correct, but maybe it would be nice to have a plan of what should be checked also. In general our tests are rather poor on negative scenarios :)

Reply:
ok , I'll describe the checks and add negative scenario(s), thanks.

Reply:
Hey, has this been resolved? I do not see much change in the PR right now 😅

Reply:
Hey, I am working on this right now, I'll let u know, thanks.

Reply:
I've tweaked the sections with some detailed description and I've added what should be checked also.

Reply:
Thank you.